evap/rewards/views.py

```diff
@@ -26,24 +27,33 @@
 from evap.staff.views import semester_view
 
 
+def redeem_reward_points(request):
+    redemptions = {}
+    try:
+        for key, value in request.POST.items():
+            if key.startswith("points-"):
+                event_id = int(key.rpartition("-")[2])
+                redemptions[event_id] = int(value)
+        previous_redeemed_points = int(request.POST["previous_redeemed_points"])
+    except (ValueError, KeyError, TypeError) as e:
+        raise BadRequest from e
+
+    try:
+        save_redemptions(request, redemptions, previous_redeemed_points)
+        messages.success(request, _("You successfully redeemed your points."))
+    except (NoPointsSelected, NotEnoughPoints, RedemptionEventExpired) as error:
+        messages.warning(request, error)
+    except OutdatedRedemptionData as error:
+        messages.error(request, error)
+        return 409
```

Comment:
While this does correctly reproduce the current behavior, it seems a bit odd that we return 200 even if saving the points failed. Let's make a "very small" issue for first semester students to make this return 400 in the first `except` instead.

I noticed that `assertContains` is a bit weird and also asserts that the status code is 200 unless told otherwise which could lead to confusion / requires us to specify this twice - I think they will handle it though :)